internal/controller/deployment/controller_finalize.go

```diff
@@ -73,12 +83,44 @@ func (r *Reconciler) finalize(ctx context.Context, old, deployment *choreov1.Dep
 	}
 
 	resourceHandlers := r.makeExternalResourceHandlers()
+	pendingDeletion := false
+
 	for _, resourceHandler := range resourceHandlers {
-		if err := resourceHandler.Delete(ctx, deploymentCtx); err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to delete external resource %s: %w", resourceHandler.Name(), err)
+		// Skip the namespace resource as it should not be considered to handle the deletion
+		if resourceHandler.Name() == k8sintegrations.NamespaceHandlerName {
+			continue
+		}
+
+		// Check if the resource is still being deleted
+		exists, err := resourceHandler.GetCurrentState(ctx, deploymentCtx)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to check existence of external resource %s: %w", resourceHandler.Name(), err)
+		}
+
+		if exists != nil {
+			pendingDeletion = true
+			// Trigger deletion of the resource as it is still exists
+			if err := resourceHandler.Delete(ctx, deploymentCtx); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to delete external resource %s: %w", resourceHandler.Name(), err)
+			}
```

Comment:
Shall we use early return here? It will reduce the nesting and keep the code more readable.

```suggestion
		if exists == nil {
		    continue
		}
		pendingDeletion = true
		// Trigger deletion of the resource as it is still exists
		if err := resourceHandler.Delete(ctx, deploymentCtx); err != nil {
			return ctrl.Result{}, fmt.Errorf("failed to delete external resource %s: %w", resourceHandler.Name(), err)
		}
```